portal/models/flaskdanceprovider.py

```diff
@@ -11,21 +11,134 @@ class FlaskDanceProvider:
     """
     __metaclass__ = ABCMeta
 
-    def __init__(self, blueprint, token):
+    def __init__(self, blueprint, token, standard_key_to_provider_key_map):
+        """constructor
+
+        :param blueprint: The provider blueprint
+        :param token: The user's access token
+        :param standard_key_to_provider_key_map: Each provider uses
+            different keys to encode user values. This maps
+            our standard keys, such as 'first_name' to provider keys,
+            such as 'given_name' in Google's case
+        """
         self.blueprint = blueprint
         self.token = token
+        self.standard_key_to_provider_key_map = \
+            standard_key_to_provider_key_map
 
     @property
     def name(self):
         return self.blueprint.name
 
-    @abstractmethod
     def get_user_info(self):
         """gets user info from the provider
 
+        This function parses json returned from the provider
+        and returns an instance of FlaskProviderUserInfo that is
+        filled with the user's information
+
+        :return FlaskProviderUserInfo with the user's info
+        """
+
+        # Ask the provider for details about the user
+        resp = self.send_get_user_json_request()
+
+        if not resp.ok:
+            current_app.logger.debug(
```

Comment:
I'm assuming this is an error/unexpected scenario?  Lets use `current_app.logger.error()` if so, so it doesn't go unnoticed.

Reply:
Not sure why I keep making this mistake :open_mouth: